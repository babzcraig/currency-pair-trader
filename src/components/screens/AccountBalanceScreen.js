import React from 'react';
import PropTypes from 'prop-types';

const AccountBalanceScreen = ({usdBalance, btcBalance}) => {
  return (
    <div className="account-balance-section">
      <h1>Account Balance</h1>
      <p>
        <span>USD :</span>{usdBalance}</p>
      <p>
        <span>BTC :</span>{btcBalance}</p>
    </div>
  )
}

AccountBalanceScreen.propTypes = {
  usdBalance: PropTypes.number,
  btcBalance: PropTypes.number
}

export default AccountBalanceScreen;
