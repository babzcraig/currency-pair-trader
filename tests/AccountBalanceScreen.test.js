import React from "react";
import ReactDOM from "react-dom";
import AccountBalanceScreen from "../src/components/screens/AccountBalanceScreen";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("<AccountBalanceScreen />", () => {
  const wrapper = shallow(
    <AccountBalanceScreen usdBalance={100} btcBalance={0.0069} />
  );

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <AccountBalanceScreen usdBalance={100} btcBalance={0.0069} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("contains a div with name account-balance-section", () => {
    expect(wrapper.find(".account-balance-section").exists()).toBe(true);
  });

  it("contains an h1 with a Text of Account Balance", () => {
    expect(wrapper.find("h1").text()).toBe("Account Balance");
  });

  it("matches the snapshot", () => {
    const tree = shallow(<AccountBalanceScreen />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
