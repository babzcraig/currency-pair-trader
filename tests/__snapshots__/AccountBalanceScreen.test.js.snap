// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountBalanceScreen /> matches the snapshot 1`] = `
<div
  className="account-balance-section"
>
  <h1>
    Account Balance
  </h1>
  <p>
    <span>
      USD :
    </span>
  </p>
  <p>
    <span>
      BTC :
    </span>
  </p>
</div>
`;
